src/Autofocus.tsx

```diff
@@ -0,0 +1,160 @@
+
+export type AutofocusProps = typeof DepthOfField & {
+  target?: [number, number, number]
+  mouse?: boolean
+  debug?: number
+  manual?: boolean
+  smoothTime?: number
+}
+
+export type AutofocusApi = {
+  dofRef: RefObject<DepthOfFieldEffect>
+  hitpoint: THREE.Vector3
+  update: (delta: number, updateTarget: boolean) => void
+}
+
+export const Autofocus = forwardRef<AutofocusApi, AutofocusProps>(
+  (
+    { target = undefined, mouse: followMouse = false, debug = undefined, manual = false, smoothTime = 0, ...props },
+    fref
+  ) => {
+    const dofRef = useRef<DepthOfFieldEffect>(null)
+    const hitpointRef = useRef<THREE.Mesh>(null)
+    const targetRef = useRef<THREE.Mesh>(null)
+
+    const { size, gl, scene } = useThree()
+    const { composer, camera } = useContext(EffectComposerContext)
+
+    // see: https://codesandbox.io/s/depthpickingpass-x130hg
+    const [depthPickingPass] = useState(new DepthPickingPass())
+    useEffect(() => {
+      const copyPass = new CopyPass()
+      composer.addPass(depthPickingPass)
+      composer.addPass(copyPass)
+      return () => {
+        composer.removePass(copyPass)
+        composer.removePass(depthPickingPass)
+      }
+    }, [composer, depthPickingPass])
+
+    const [hitpoint] = useState(new THREE.Vector3(0, 0, 0))
+
+    const [ndc] = useState(new THREE.Vector3(0, 0, 0))
+    const getHit = useCallback(
+      async (x: number, y: number) => {
+        ndc.x = x
+        ndc.y = y
+        ndc.z = await depthPickingPass.readDepth(ndc)
+        ndc.z = ndc.z * 2.0 - 1.0
+        const hit = 1 - ndc.z > 0.0000001 // it is missed if ndc.z is close to 1
+        return hit ? ndc.unproject(camera) : false
+      },
+      [ndc, depthPickingPass, camera]
+    )
+
+    const [pointer] = useState(new THREE.Vector2())
+    useEffect(() => {
+      if (!followMouse) return
+
+      async function onPointermove(e: PointerEvent) {
+        const clientX = e.clientX - size.left
+        const clientY = e.clientY - size.top
+        const x = (clientX / size.width) * 2.0 - 1.0
+        const y = -(clientY / size.height) * 2.0 + 1.0
+
+        pointer.set(x, y)
+      }
+      gl.domElement.addEventListener('pointermove', onPointermove, {
+        passive: true,
+      })
+
+      return () => void gl.domElement.removeEventListener('pointermove', onPointermove)
+    }, [gl.domElement, hitpoint, size, followMouse, getHit, pointer])
+
+    const update = useCallback(
+      async (delta: number, updateTarget = true) => {
+        // Update hitpoint
+        if (target) {
+          hitpoint.set(...target)
+        } else {
+          const { x, y } = followMouse ? pointer : { x: 0, y: 0 }
+          const hit = await getHit(x, y)
+          if (hit) hitpoint.copy(hit)
+        }
+
+        // Update target
+        if (updateTarget && dofRef.current?.target) {
+          if (smoothTime > 0 && delta > 0) {
+            easing.damp3(dofRef.current.target, hitpoint, smoothTime, delta)
```

Comment:
@drcmda damping target is here directly supported by the component, but as mentioned in the doc, the ref-api allows doing it from the outside.: https://github.com/pmndrs/react-postprocessing/blob/6f733b2a469c79bf8de2bb2b877455e256a476d5/docs/effects/autofocus.mdx?plain=1#L40-L48

If you prefer, I can remove the `maath` dep and remove damping feature, but I think it is handy/user-friendly to have it by default.

Let me know ;)

Reply:
maath is good imo